clippy_dev/src/new_lint.rs

```diff
@@ -0,0 +1,184 @@
+use std::fs::{File, OpenOptions};
+use std::io;
+use std::io::prelude::*;
+use std::io::ErrorKind;
+use std::path::{Path, PathBuf};
+
+pub fn create(pass: Option<&str>, lint_name: Option<&str>, category: Option<&str>) -> Result<(), io::Error> {
+    let pass = pass.expect("`pass` argument is validated by clap");
+    let lint_name = lint_name.expect("`name` argument is validated by clap");
+    let category = category.expect("`category` argument is validated by clap");
+
+    match open_files(lint_name) {
+        Ok((mut test_file, mut lint_file)) => {
+            let (pass_type, pass_import, context_import) = match pass {
+                "early" => ("EarlyLintPass", "use syntax::ast::*;", "EarlyContext"),
+                "late" => ("LateLintPass", "use rustc_hir::*;", "LateContext"),
+                _ => {
+                    unreachable!("`pass_type` should only ever be `early` or `late`!");
+                },
+            };
+
+            let camel_case_name = to_camel_case(lint_name);
+
+            match test_file.write_all(get_test_file_contents(lint_name).as_bytes()) {
+                Ok(()) => (),
+                Err(e) => {
+                    eprintln!("Could not write to test file: {}", e);
+                    return Err(io::Error::new(
+                        ErrorKind::Other,
+                        format!("Could not write to test file: {}", e),
+                    ));
+                },
+            };
+
+            match lint_file.write_all(
+                get_lint_file_contents(
+                    pass_type,
+                    lint_name,
+                    &camel_case_name,
+                    category,
+                    pass_import,
+                    context_import,
+                )
+                .as_bytes(),
+            ) {
+                Ok(()) => Ok(()),
+                Err(e) => Err(io::Error::new(
+                    ErrorKind::Other,
+                    format!("Could not write to lint file: {}", e),
+                )),
+            }
+        },
+        Err(e) => Err(io::Error::new(
+            ErrorKind::Other,
+            format!("Unable to create lint: {}", e),
+        )),
+    }
+}
+
+fn open_files(lint_name: &str) -> Result<(File, File), io::Error> {
+    let project_root = project_root()?;
+
+    let test_file_path = project_root.join(format!("tests/ui/{}.rs", lint_name));
+    let lint_file_path = project_root.join(format!("clippy_lints/src/{}.rs", lint_name));
```

Comment:
This makes problems with windows. Can you build the paths with `PathBuf` or `join` each dir on its own?

Reply:
done